web/modules/custom/weather_block/weather_block.services.yml

```diff
@@ -0,0 +1,4 @@
+services:
+  weather_block.save_cities_for_user:
+    class: Drupal\weather_block\Services\GetSetCityUser
```

Comment:
Weird namings.
save_citiest_for_user
but class name suggest it also can get city user
and service itself also could fetch data from API.

Service should be responsible for some single set of functionality. For example with these names I would never think that it will also be responsible for weather API

Reply:
It is not much better to "include every method" in the name
Just name it UserWeatherHandler or whatever.
In other words make sure it describes functionality with assumption that in future you will be able to add more method that still related to that functionality without making the service name obsolete